assignment/impl/src/java/org/sakaiproject/assignment/impl/AssignmentServiceImpl.java

```diff
@@ -1403,32 +1403,58 @@ public void updateAssignment(Assignment assignment) throws PermissionException {
             throw new PermissionException(sessionManager.getCurrentSessionUserId(), SECURE_UPDATE_ASSIGNMENT, null);
         }
 
-        Collection<String> oldGroups = assignmentRepository.findGroupsForAssignmentById(assignment.getId());
+        Collection<String> prevAssignedGroupRefs = assignmentRepository.findGroupsForAssignmentById(assignment.getId());
         switch (assignment.getTypeOfAccess()) {
             case GROUP:
-                oldGroups.removeAll(assignment.getGroups());
-                for (String groupRef : oldGroups) { // remove locks for groups that were removed
-                    try {
-                        AuthzGroup group = authzGroupService.getAuthzGroup(groupRef);
-                        group.setLockForReference(reference, AuthzGroup.RealmLockMode.NONE);
-                        authzGroupService.save(group);
-                    } catch (GroupNotDefinedException | AuthzPermissionException e) {
-                        log.warn("Exception while removing lock for assignment {}, {}", assignment.getId(), e.toString());
+                Collection<String> currentAssignedGroupRefs = assignment.getGroups();
+                // get the unassigned Group References by removing the current ones !
+                prevAssignedGroupRefs.removeAll(currentAssignedGroupRefs);
+
+                Site site = null;
+                try {
+                    site = siteService.getSite(assignment.getContext());
+                } catch (IdUnusedException e) {
+                    log.warn("Could not get the site {} for assignment {} while updating it", assignment.getContext(), assignment.getId());
+                }
+                // on Lessons, a group assignment or one with prereq. will be managed through a single (access) group
+                Group anAssignedGroup = site.getGroup(currentAssignedGroupRefs.stream().findFirst().orElse(null));
+                // we can infer it is an ACCESS group if it has the following property
+                String assignedGroupLessonRef = anAssignedGroup.getProperties().getProperty("lessonbuilder_ref");
```

Comment:
boolean accessGroupAssigned = anAssignedGroup.getProperties().getProperty("lessonbuilder_ref") != null ? true : false;

Alternative 👆🏼

Reply:
Dafuq!

boolean accessGroupAssigned = anAssignedGroup.getProperties().getProperty("lessonbuilder_ref") != null;

Reply:
Would be .isNotBlank for this case, thanks!